libsimulator/src/Ellipse.cpp

```diff
@@ -3,19 +3,58 @@
 #include "Ellipse.hpp"
 #include "Macros.hpp"
 
-// ellipse  semi-axis in the direction of the velocity
+/**
+ * @brief Calculates the semi-major axis (EA) of an ellipse based on the given speed.
+ *
+ * The ellipse adapts dynamically depending on the agent's speed. This method computes
+ * the semi-axis length in the direction of movement, reflecting how the ellipse
+ * elongates as speed increases.
+ *
+ * @param speed The current speed of the object (must be non-negative).
+ * @return The computed semi-axis length in the velocity direction.
+ *
+ */
```

Comment:
Please follow the established documentation style and use /// to indicate doc-style comments.